### PR TITLE
:tada: New Source: Polygon Stocks

### PR DESCRIPTION
**Pull request by Harshil0512**
## What
The Following PR adds a Low Code CDK Source for Polygon-Stocks-API provided by https://polygon.io/

## How
The PR uses the configuration based CDK and implements full refresh syncs of the following streams :
- stock_api_stream

## Recommended reading order
1. `spec.yaml`
2. `polygon_stock_api.yaml`
3. `streams`

## 🚨 User Impact 🚨
No Changes To Existing Code

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [x] *Community member?* Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [x] Documentation updated
    - [x] Connector's `README.md`
    - [x] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [x] `docs/integrations/README.md`
    - [ x `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [x] Create a non-forked branch based on this PR and test the below items on it
- [x] Build is successful
- [x] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [x] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [x] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [x] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [x] Documentation updated
    - [x] Connector's `README.md`
    - [x] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [x] Create a non-forked branch based on this PR and test the below items on it
- [x] Build is successful
- [x] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [x] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [x] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [x] Issue acceptance criteria met
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [x] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [x] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [x] Documentation which references the generator is updated as needed

</details>

## Tests
Polygon_Stocks_API

<details><summary><strong>Unit</strong></summary>

Put your unit tests output here.

</details>

<details><summary><strong>Integration</strong></summary>

Put your integration tests output here.

</details>

<details><summary><strong>Acceptance</strong></summary>

Put your acceptance tests output here.

</details>

### CODE REVIEW

**Comment by Harshil0512** [comment]
Issue related to this PR : 

https://github.com/airbytehq/connector-contest/issues/261

**Comment by marcosmarxm** [comment]
@Harshil0512 please check your commits there is something wrong with it.

**Comment by Harshil0512** [comment]
@marcosmarxm may I know what is wrong with the commit ? is there something wrong with the files which I created or something with the files belongs to your repository updated due to my commit?

**Comment by Harshil0512** [comment]
Sir I have worked hard for this so kindly look once in the weather api we have features of finding weather by langitude ,longitude ,city,etc.

**Comment by marcosmarxm** [comment]
You're pushing files that don't make part of the pull request goal. Take a look.

**Comment by Harshil0512** [comment]
Okay let me see and update it

**Comment by Harshil0512** [comment]
@marcosmarxm can you please check the files now?

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Hello @Harshil0512, Marcos from Airbyte here 👋 . We received more than 25 new contributions along the weekend. One is yours 🎉  thank so much for! Our team is limited and maybe the review process can take longer than expected. As described in the Airbyte's Hacktoberfest your contribution was submitted before November 2nd and it is eligible to win the prize. The review process will validate other requirements. I ask to you patience until someone from the team review it. 

Because I reviewed some contributions for Hacktoberfest so far I saw some common patterns you can check in advance:
* [ ] Make sure you have added connector documentation to `/docs/integrations/`
* [ ] Remove the file `catalog` from `/integration_tests`
* [ ] Edit the `sample_config.json` inside `/integration_tests`
* [ ] For the `configured_catalog` you can use only `json_schema: {}`
* [ ] Add title to all properties in the `spec.yaml`
* [ ] Make sure the `documentationUrl` in the `spec.yaml` redirect to Airbyte's future connector page, eg: connector Airtable the `documentationUrl: https://docs.airbyte.com/integrations/sources/airtable`
* [ ] Review now new line at EOF (end-of-file) for all files.

If possible send to me a DM in Slack with the tests credentials, this process will make easier to us run integration tests and publish your connector. If you only has production keys, make sure to create a bootstrap.md explaining how to get the keys.

**Comment by Harshil0512** [comment]
Okat let me review the files once again. Thank you.

**Comment by marcosmarxm** [comment]
Hello! I'm going to be out of the office this Friday and won't be able to review your contribution again today, I return next Monday. So far, most contributions look solid and are almost done to be approved. As said in [Chris' comment](https://airbytehq.slack.com/archives/C0450PCSDB2/p1667430678866629) all contributions made before 2-November are eligible to receive the prize and have 2 weeks to merge the contributions. But I ensure next week we're going to have your contribution merged. If you have questions about the implementation you can send them in `#hacktoberfest-2022` in Airbyte's Slack.

Sorry the inconvenience and see you again next week, thank you so much for your contribution!

**Comment by Harshil0512** [comment]
Sure No Problem @marcosmarxm

**Comment by Harshil0512** [comment]
I performed the changes mentioned in the review changes

**Comment by Harshil0512** [comment]
> Hello! I'm going to be out of the office this Friday and won't be able to review your contribution again today, I return next Monday. So far, most contributions look solid and are almost done to be approved. As said in [Chris' comment](https://airbytehq.slack.com/archives/C0450PCSDB2/p1667430678866629) all contributions made before 2-November are eligible to receive the prize and have 2 weeks to merge the contributions. But I ensure next week we're going to have your contribution merged. If you have questions about the implementation you can send them in `#hacktoberfest-2022` in Airbyte's Slack.
> 
> Sorry the inconvenience and see you again next week, thank you so much for your contribution!


Are you there @marcosmarxm

File: airbyte-integrations/connectors/source-polygon-stock-api/source_polygon_stock_api/polygon_stock_api.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      $ref: "*ref(definitions.requester)"
+  base_stream:
+    retriever:
+      $ref: "*ref(definitions.retriever)"
+  stock_api_stream:
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "stock_api"
+      primary_key: "id"
+      path: "/v2/aggs/ticker/{{ config['stocksTicker'] }}/range/{{ config['multiplier'] }}/{{ config['timespan'] }}/{{ config['from'] }}/{{ config['to'] }}?adjusted={{ config['adjusted'] }}&sort={{ config['sort'] }}&limit=120&apiKey={{ config['apiKey'] }}"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-polygon-stock-api/source_polygon_stock_api/polygon_stock_api.yaml:29]: Please overwrite the request parameter component. 
Please implement the DateStreamSlicer to enable incremental syncs.
Always sort by desc to read first old to newest data.

File: airbyte-integrations/connectors/source-polygon-stock-api/source_polygon_stock_api/polygon_stock_api.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+streams:
+  - "*ref(definitions.stock_api_stream)"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-polygon-stock-api/source_polygon_stock_api/polygon_stock_api.yaml:31]: Are you going to add only this endpoint? I saw other endpoints in the API docs.

File: airbyte-integrations/connectors/source-polygon-stock-api/source_polygon_stock_api/spec.yaml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    from:
+      type: string
+      description: The beginning date for the aggregate window.
+      examples:
+        - "2020-10-14"
+    to:
+      type: string
+      description: The target date for the aggregate window.
+      examples:
+        - "2020-10-14"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-polygon-stock-api/source_polygon_stock_api/spec.yaml:45]: Change `from` to `start_date` and `to` to `end_date` is more used by Airbyte's connectors.

File: airbyte-integrations/connectors/source-polygon-stock-api/source_polygon_stock_api/spec.yaml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      description: Your API ACCESS Key
+      airbyte_secret: true
+    stocksTicker:
+      type: string
+      description: The exchange symbol that this item is traded under.
+      examples:
+        - IBM
+        - MSFT
+    multiplier:
+      type: integer
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-polygon-stock-api/source_polygon_stock_api/spec.yaml:26]: Add title to your parameters.

File: airbyte-integrations/connectors/source-polygon-stock-api/source_polygon_stock_api/spec.yaml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  $schema: http://json-schema.org/draft-07/schema#
+  title: Weather API Spec
+  type: object
+  required:
+    - apiKey
+    - stocksTicker
+    - multiplier
+    - timespan
+    - from
+    - to
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-polygon-stock-api/source_polygon_stock_api/spec.yaml:12]: `to` or `end_date` is not required if you implement the datestream slicer.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Some comments.

**Comment by Harshil0512** [comment]
Okay I'll work on that and will keep you updated

File: airbyte-integrations/connectors/source-polygon-stock-api/source_polygon_stock_api/spec.yaml (hunk 3 above)
Comment (Harshil0512) [reply airbyte-integrations/connectors/source-polygon-stock-api/source_polygon_stock_api/spec.yaml]: I did this

File: airbyte-integrations/connectors/source-polygon-stock-api/source_polygon_stock_api/polygon_stock_api.yaml (hunk 2 above)
Comment (Harshil0512) [reply airbyte-integrations/connectors/source-polygon-stock-api/source_polygon_stock_api/polygon_stock_api.yaml:38]: Yes as I thought it is the most important end point as per my point of view

File: airbyte-integrations/connectors/source-polygon-stock-api/source_polygon_stock_api/polygon_stock_api.yaml (hunk 1 above)
Comment (Harshil0512) [reply airbyte-integrations/connectors/source-polygon-stock-api/source_polygon_stock_api/polygon_stock_api.yaml]: This is still to be done I'll finish this soon.

File: airbyte-integrations/connectors/source-polygon-stock-api/source_polygon_stock_api/spec.yaml (hunk 4 above)
Comment (Harshil0512) [reply airbyte-integrations/connectors/source-polygon-stock-api/source_polygon_stock_api/spec.yaml:29]: I did this

**Comment by marcosmarxm** [comment]
@Harshil0512 I had a problem when pushing some change to your branch :( I'll finish your work at #19504 Thanks so much for the contribution

@RealChrisSean same problem real, sorry!!!

**Pull request closed without merging**